libqpdf/QPDFWriter.cc

```diff
@@ -3154,3 +3199,88 @@ QPDFWriter::writeStandard()
         assert(this->md5_pipeline == 0);
     }
 }
+
+void
+QPDFWriter::writePCLm()
+{
+    if (this->deterministic_id)
+    {
+        pushMD5Pipeline();
+    }
+
+    // Start writing
+
+    writePCLmHeader();
+    writeString(this->extra_header_text);
+
+    // Image transform stream content for page strip images.
+    // Each of this new stream has to come after every page image
+    // strip written in the pclm file.
+    std::string image_transform_content = "q /image Do q\n";
+
+    // enqueue all pages first
+    std::vector<QPDFObjectHandle> all = this->pdf.getAllPages();
+    for (std::vector<QPDFObjectHandle>::iterator iter = all.begin();
+            iter != all.end(); ++iter)
+    {
+        // enqueue page
+        enqueueObject(*iter);
+
+        // enqueue page contents stream
+        enqueueObject((*iter).getKey("/Contents"));
+
+        // enqueue all the strips for each page
+        QPDFObjectHandle strips = (*iter).getKey("/Resources").getKey("/XObject");
+        std::set<std::string> keys = strips.getKeys();
+        for (std::set<std::string>::iterator image = keys.begin();
+                image != keys.end(); ++image)
+        {
+            enqueueObject(strips.getKey(*image));
+            enqueueObject(QPDFObjectHandle::newStream(&pdf, image_transform_content));
```

Comment:
In a PDF, the page resources object looks like: 
```
 /Resources << /XObject << /I 5 0 R >> >> 
``` 
In PCLm, they are strips, similar to:
```
/Resources << /XObject << /Image0 3 0 R /Image1 5 0 R /Image2 7 0 R /Image3 9 0 R >> >>
```

In between two objects in the above PCLm example, we need the following image transform stream:
```
4 0 obj
<< /Length 14 >>
stream
q /image Do Q
endstream
endobj
```
Since the contents of this stream will always be the same, and equal to the string `q /image Do Q` according to the PCLm specifications, I though it will be better if I directly add them to where they should be. Does this make sense?